### PR TITLE
Add build scripts for sqlite-vec

### LINKED ISSUE #215: [Request]: Upgrade sqlitevec.xcframework
### What do you need?

It seems like the sqlite-vec binding is not the [latest version](https://github.com/asg017/sqlite-vec/releases/tag/v0.1.6), can you upgrade that? Thanks!

Or, if you like to have a build script in repo, I'd be happy to try to contribute.

### PR DESCRIPTION
**Pull request by jhen0409**
Fix #215.

Build framework for iOS, and download Android libs from sqlite-vec [releases](https://github.com/asg017/sqlite-vec/releases/tag/v0.1.6).

### CODE REVIEW

File: scripts/build-sqlite-vec-ios.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/bin/sh -e
+
+SQLITEVEC_VERSION=0.1.6
```
Comment (ospfranco) [inline scripts/build-sqlite-vec-ios.sh:3]: Can you make this a cli argument?

File: scripts/download-sqlite-vec-android.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  local download_url="https://github.com/asg017/sqlite-vec/releases/download/v$SQLITEVEC_VERSION/sqlite-vec-$SQLITEVEC_VERSION-loadable-android-$arch.tar.gz"
+  if [ ! -f sqlite-vec-$SQLITEVEC_VERSION-loadable-android-$arch.tar.gz ]; then
+    wget -O sqlite-vec-$SQLITEVEC_VERSION-loadable-android-$arch.tar.gz $download_url
+    tar -xvf sqlite-vec-$SQLITEVEC_VERSION-loadable-android-$arch.tar.gz -C ../android/src/main/jniLibs/$abi/
+    mv ../android/src/main/jniLibs/$abi/vec0.so ../android/src/main/jniLibs/$abi/libsqlite_vec.so
+    rm sqlite-vec-$SQLITEVEC_VERSION-loadable-android-$arch.tar.gz
+  fi
+}
+
+download_sqlite_vec arm64-v8a aarch64
```
Comment (ospfranco) [inline scripts/download-sqlite-vec-android.sh:17]: Can you combine both scripts into a single one?

**Review by ospfranco — COMMENTED** [review]
Have you tested the binaries generated by the script?

**Comment by jhen0409** [comment]
> Have you tested the binaries generated by the script?

I've tested to replace binaries in my project, but only tested on iOS arm64 simulator and Android arm64-v8a device.

Three lines to validate it works:

```js
await db.execute(`CREATE VIRTUAL TABLE vec_items USING vec0(txt TEXT, embedding float[768]);`)
await db.execute(
  `INSERT INTO vec_items(rowid, txt, embedding) VALUES (?, ?, ?)`,
  id, text, new Float32Array(embeddings)],
)
console.log(await db.execute(`SELECT * FROM vec_items`)
```

**Comment by ospfranco** [comment]
Awesome, thanks!

**Pull request merged**